src/Protocol/ATProtocol/Actor.php

```diff
@@ -174,27 +174,24 @@ public function getContactByDID(string $did, int $uid, int $contact_uid): array
 			return $contact;
 		}
 
-		if (empty($contact)) {
-			$fields = [
-				'uid'      => $contact_uid,
-				'network'  => Protocol::BLUESKY,
-				'priority' => 1,
-				'writable' => true,
-				'blocked'  => false,
-				'readonly' => false,
-				'pending'  => false,
-				'url'      => $did,
-				'nurl'     => $did,
-				'alias'    => ATProtocol::WEB . '/profile/' . $did,
-				'name'     => $did,
-				'nick'     => $did,
-				'addr'     => $did,
-				'rel'      => Contact::NOTHING,
-			];
-			$cid = Contact::insert($fields);
-		} else {
-			$cid = $contact['id'];
-		}
+		$fields = [
+			'uid'      => $contact_uid,
+			'network'  => Protocol::BLUESKY,
+			'priority' => 1,
+			'writable' => true,
+			'blocked'  => false,
+			'readonly' => false,
+			'pending'  => false,
+			'url'      => $did,
+			'nurl'     => $did,
+			'alias'    => ATProtocol::WEB . '/profile/' . $did,
+			'name'     => $did,
+			'nick'     => $did,
+			'addr'     => $did,
+			'rel'      => Contact::NOTHING,
+		];
+
+		$cid = Contact::insert($fields);
```

Comment:
We mustn't insert the contact when the contact already exists. Also we really shouldn't overwrite the `rel` value, since that indicates the relationship between the contact and the user.

Reply:
Oh, you are right. That is definitely a fragment from the transition from the addon code to the core code. I will most likely have to apply some change there, when I touch the addon.